### PR TITLE
HtmlHelper.DisplayTextFor should use DisplayAttribute of enums

### PR DESCRIPTION
**Pull request by dougbu**
Continue on from #7149 with a few more tests
- includes @mistakenot's original #7033 though squashed and rebased

### CODE REVIEW

**Comment by dougbu** [comment]
The second commit addresses my [comment](https://github.com/aspnet/Mvc/pull/7149/files#r166098692) in #7149.

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ModelExplorerExtensions.cs [hunk 1]
```diff
@@ -40,6 +41,22 @@ public static string GetSimpleDisplayText(this ModelExplorer modelExplorer)
                 return modelExplorer.Metadata.NullDisplayText;
             }
 
+            if (modelExplorer.Metadata.IsEnum && modelExplorer.Model is Enum modelEnum)
+            {
+                var enumStringValue = modelEnum.ToString("d");
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ModelExplorerExtensions.cs]: Should this specify `InvariantCulture`?

Comment (dougbu) [reply src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ModelExplorerExtensions.cs]: The `Enum.ToString(...)` overloads taking an `IFormatProvider` are both `[Obsolete]`.

**Comment by dougbu** [comment]
- 4f3e044928
- daf341df35

Thanks @mistakenot!

**Pull request closed without merging**